src/compiler/checker.ts

```diff
@@ -22005,6 +22005,19 @@ namespace ts {
                     }
                 }
             }
+
+            // If location is stringLiteral, it might be argument in ElementAccessExpression, like a["Q"].
+            // Deal it just like identifier/PropertyAccessExpression.
+            if (location.kind === SyntaxKind.StringLiteral) {
+                const candidateExpression = location.parent;
+                if (isElementAccessExpression(candidateExpression) && !isAssignmentTarget(candidateExpression)) {

```

Comment:
You're not actually checking whether this is the argument of the `ElementAccessExpression`. For all you know, the expression could be

```ts
"hello"[someIdentifier]
```

It probably doesn't matter, but I'd rather you added the check in to see if `candidateExpression.argumentExpression === location`.